File: Solutions/Easy/2273. Find Resultant Array After Removing Anagrams/Solution.py
class Solution:
    def removeAnagrams(self, words: list[str]) -> list[str]:
        if len(words) == 1:
            return words
        res: list[str] = [words[0]]
        sortedString: str = ''.join(sorted(words[0]))
        prev: str = sortedString
        for i in range(1, len(words)):
            sortedString = ''.join(sorted(words[i]))
            if (prev != sortedString):
                res.append(words[i])
                prev = sortedString
        return res

File: Solutions/Easy/2273. Find Resultant Array After Removing Anagrams/Solution.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Solution {
    public IList<string> RemoveAnagrams(string[] words) {
        if(words.Length == 1) return words.ToList();
        List<string> res = new List<string>();
        res.Add(words[0]);
        char[] arr = words[0].ToCharArray();
        Array.Sort(arr);
        string prev = new string(arr);
        for(int i = 1; i < words.Length; i++){
            arr = words[i].ToCharArray();
            Array.Sort(arr);
            string sorted = new string(arr);
            if(prev != sorted){
                prev = sorted;
                res.Add(words[i]);
            }
        }
        return res;
    }
}
